src/main/java/randoop/operation/TypedOperation.java

```diff
@@ -70,6 +84,83 @@ public abstract class TypedOperation implements Operation, Comparable<TypedOpera
     this.inputTypes = inputTypes;
     this.outputType = outputType;
     this.execSpec = execSpec;
+    if (operation.isMethodCall()) {
+      MethodCall methodCall = (MethodCall) operation;
+      Method m = methodCall.getMethod();
+      Class<?> declaringClass = m.getDeclaringClass();
+      String className = declaringClass.getName();
+      OperationSpecification OperationSpec = specificationFromAnnotations(m, className);
+      try (SequenceCompiler compiler = new SequenceCompiler()) {
+        // Use the createExecutableSpecification method
+        ExecutableSpecification annoSpec =
+            SpecificationTranslator.createExecutableSpecification(m, OperationSpec, compiler);
+        if (this.execSpec == null) {
+          this.execSpec = annoSpec;
+        } else {
+          this.execSpec = ExecutableSpecification.merge(annoSpec, this.execSpec);
+        }
+      } catch (Exception e) {
+        System.out.println("Exception occurred while creating and merging ExecutableSpecification");
+      }
+    }
+  }
+
+  /**
+   * Creates a specification for the given method based on Nonnull annotations for parameter and
+   * return types.
+   *
+   * @param method the method to extract the specification from
+   * @param className the name of the class that declares the method
+   * @return the specification for the method
+   */
+  private static OperationSpecification specificationFromAnnotations(
+      Method method, @ClassGetName String className) {
+    String methodName = method.getName();
+    List<String> parameterNames =
+        CollectionsPlume.mapList(Parameter::getName, method.getParameters());
+    List<@ClassGetName String> parameterTypes =
+        CollectionsPlume.mapList(Class::getName, method.getParameterTypes());
+
+    OperationSignature operation =
+        OperationSignature.forConstructorName(className, methodName, parameterTypes);
+    Identifiers identifiers = new Identifiers("receiver", parameterNames, "result");
```

Comment:
Is a `result` identifier always needed?  What if the method has `void` return type?

Reply:
The Identifiers class should assign a "result" identifier in it's constructor regardless of the return type. If the return type is void I do not think the method would have a pure annotation.